src/common/hist_util.cu

```diff
@@ -208,7 +221,8 @@ void ExtractWeightedCuts(int device, Span<SketchEntry> cuts,
 void ProcessBatch(int device, const SparsePage& page, size_t begin, size_t end,
                   SketchContainer* sketch_container, int num_cuts,
                   size_t num_columns) {
-  dh::XGBCachingDeviceAllocator<char> alloc;
+  dh::XGBCachingDeviceAllocator<char> caching_alloc;
+  dh::XGBDeviceAllocator<char> alloc;
```

Comment:
should we get rid of this and use the caching_alloc throughout?

Reply:
Yes